### PR TITLE
support azure llm deployment

### PR DESCRIPTION
**Pull request by zycarlc**
## please note I accidently close my pull request #847 , this is a re-summit pull request.

This pull request includes changes to add support for Azure LLM models and update configuration files accordingly. The most important changes include modifications to the README.md, config.yaml, and the llm_manager.py file to accommodate the new Azure model type.

Addressing the feature request below
[FEATURE]: Add Azure OpenAI https://github.com/AIHawk-co/Auto_Jobs_Applier_AI_Agent/issues/616

### CODE REVIEW

**Comment by 49Simon** [comment]
@zycarlc Kindly resolve conflicts first and I can proceed with reviewing your PR

**Comment by zycarlc** [comment]
> @zycarlc Kindly resolve conflicts first and I can proceed with reviewing your PR

Resolved as requested.

File: data_folder/work_preferences.yaml [hunk 1]
```diff
@@ -48,4 +48,8 @@ location_blacklist:
 
 job_applicants_threshold:
   min_applicants: 0
-  max_applicants: 30
\ No newline at end of file
+  max_applicants: 30
+
+llm_model_type: openai
+llm_model: "gpt-4o-mini"
```
Comment (49Simon) [inline data_folder/work_preferences.yaml:54]: No need to keep `llm_model_type` and `llm_model` here. They have been moved config.py

File: src/ai_hawk/llm/llm_manager.py [hunk 2]
```diff
@@ -232,6 +250,10 @@ def _create_model(self, config: dict, api_key: str) -> AIModel:
             return OllamaModel(llm_model, llm_api_url)
         elif llm_model_type == GEMINI:
             return GeminiModel(api_key, llm_model)
+        elif llm_model_type == "huggingface":
+            return HuggingFaceModel(api_key, llm_model)
+        elif llm_model_type == "azure":
```
Comment (49Simon) [inline src/ai_hawk/llm/llm_manager.py:255]: Can you move the strings "azure" and "huggingface" to the constants.py file? Make it consistent with the others.

File: README.md [hunk 3]
```diff
@@ -195,371 +206,407 @@ This file contains sensitive information. Never share or commit this file to ver
...
+-   `LLM_MODEL`:
+    -   Choose the LLM model, currently supported:
+        -   openai: gpt-4o
+        -   ollama: llama2, mistral:v0.3
+        -   claude: any model
+        -   gemini: any model
+        -   azure: leave blank. your model deployment below define the model
+        -   groq: llama3-groq-70b-8192-tool-use-preview, llama3-groq-8b-8192-tool-use-preview, llama-3.1-70b-versatile, llama-3.1-8b-instant, llama-3.2-3b-preview, llama3-70b-8192, llama3-8b-8192, mixtral-8x7b-32768
+        -   aiml: any model
+-   `llm_api_url`:
```
Comment (49Simon) [inline README.md:299]: Can you change this to upper case to make it consistent with the variables in the config.py file? Do that for the other variables below as well.

**Review by 49Simon — CHANGES_REQUESTED** [review]
The changes in the README file are a bit extensive. Other than the azure deployment related, are the changes done by some linter?

File: src/ai_hawk/llm/llm_manager.py [hunk 4]
```diff
@@ -232,6 +250,10 @@ def _create_model(self, config: dict, api_key: str) -> AIModel:
             return OllamaModel(llm_model, llm_api_url)
         elif llm_model_type == GEMINI:
             return GeminiModel(api_key, llm_model)
+        elif llm_model_type == "huggingface":
```
Comment (49Simon) [inline src/ai_hawk/llm/llm_manager.py:253]: This is already defined below. You can remove it.

**Comment by zycarlc** [comment]
> The changes in the README file are a bit extensive. Other than the azure deployment related, are the changes done by some linter?

I have fixed this issue. Yes it was. It would be great if the project had the lint config file

File: data_folder/work_preferences.yaml (hunk 1 above)
Comment (zycarlc) [reply data_folder/work_preferences.yaml]: Done

File: src/ai_hawk/llm/llm_manager.py (hunk 2 above)
Comment (zycarlc) [reply src/ai_hawk/llm/llm_manager.py]: Done

File: README.md (hunk 3 above)
Comment (zycarlc) [reply README.md]: Done

File: src/ai_hawk/llm/llm_manager.py (hunk 4 above)
Comment (zycarlc) [reply src/ai_hawk/llm/llm_manager.py]: Sure

**Comment by feder-cr** [comment]
@zycarlc resolve the conflicts please

**Comment by feder-cr** [comment]
@zycarlc ?

**Pull request closed without merging**